app/questionnaire/value_source_resolver.py

```diff
@@ -100,36 +114,61 @@ def _resolve_list_item_id_for_value_source(
             else None
         )
 
+    def _resolve_repeating_answers_for_list(
+        self, *, answer_id: str, list_name: str
+    ) -> ResolvedAnswerList:
+        """Return the list of answers in answer store that correspond to the given list name and dynamic/repeating answer_id"""
+        answer_values: ResolvedAnswerList = []
+        for list_item_id in self.list_store[list_name]:
+            answer_value = self._get_answer_value(
+                answer_id=answer_id, list_item_id=list_item_id
+            )
+            if answer_value is not None and self.escape_answer_values:
+                answer_values.append(escape_answer_value(answer_value))
+            else:
+                answer_values.append(answer_value)
```

Comment:
With this logic, the else values might contain `None` if self.escape is false. Both should return same list, 1 should just be escaped.

Reply:
> would there be any other use for resolving a list where it might be relevant to include the Nones

I can't think of any. Possibly in the future.

> it occured to me the original _resolve_answer_value_source would return none if the answer_value is None

That makes sense given we have to return something, None being the default, but when returning a list of items, i think it is fine to only return if we have a value. We haven't got a need for the other case, and given the existing logic for resolving list was only adding to the list if we had an answer (even though it was naively checking the truthiness since it would treat 0 as invalid), it only returned if it has a value.

Reply:
Yeah that makes sense to me - will update 👍